pandas/core/internals/array_manager.py

```diff
@@ -999,11 +1000,16 @@ def _reindex_indexer(
 
         else:
             validate_indices(indexer, len(self._axes[0]))
+            indexer = ensure_int64(indexer)
+            if (indexer == -1).any():
+                allow_fill = True
+            else:
+                allow_fill = False
```

Comment:
@jorisvandenbossche in looking at the benchmarks you posted in #39146 it seems like some of them involve optimizations implemented for AM that _could_ be implemented for BM but weren't.  is there anything here that falls into that category?

Reply:
> in looking at the benchmarks you posted in #39146 it seems like some of them involve optimizations implemented for AM that _could_ be implemented for BM but weren't

Are you thinking of something specific?

The optimization I do here above _can_ probably be useful if you have lots of non-consolidated blocks, but for the benchmark cases with single blocks posted in #39146, that wouldn't make a difference.